File: frontend/src/mocks/admin/mcp.ts
/* ============================================
   Mock data — Admin MCP
   Extracted from features/admin/mcp/AdminMcpPage.tsx
   ============================================ */

import type { AdminMcpItem, McpRecommendItem, PaginatedResponse } from '../types'
import { listDelay, itemDelay, mutationDelay, heavyDelay } from '../delay'

export const PAGE_SIZE = 10

/* ============================================
   In‑memory store
   ============================================ */

let mcpStore: AdminMcpItem[] = []
let storeInitialized = false

function ensureStore(): AdminMcpItem[] {
  if (!storeInitialized) {
    mcpStore = generateMockMcps()
    storeInitialized = true
  }
  return mcpStore
}

function nextId(): number {
  const store = ensureStore()
  const ids = store.map((m) => m.mcpId)
  return Math.max(0, ...ids) + 1
}

/* ============================================
   Mock data generators
   ============================================ */

function generateMockMcps(): AdminMcpItem[] {
  const now = Date.now()
  return [
    {
      mcpId: 1,
      name: 'filesystem',
      displayName: '文件系统',
      icon: 'folder',
      description: '访问本地文件系统，读写文件',
      transport: 'Stdio',
      httpUrl: '',
      httpHeader: '',
      httpProxyUrl: '',
      stdioType: 'npx',
      stdioEnv: '',
      stdioArgs: '["@modelcontextprotocol/server-filesystem","/tmp"]',
      status: 1,
      healthLatency: 45,
      healthCheckedAt: new Date(now - 60000).toISOString(),
      remark: '',
      created: '2025-06-01T08:00:00Z',
      updated: '2025-06-10T14:30:00Z',
    },
    {
      mcpId: 2,
      name: 'github',
      displayName: 'GitHub',
      icon: 'git-branch',
      description: '管理 GitHub 仓库、Issues、PR',
      transport: 'Stdio',
      httpUrl: '',
      httpHeader: '',
      httpProxyUrl: '',
      stdioType: 'npx',
      stdioEnv: '',
      stdioArgs: '["@modelcontextprotocol/server-github"]',
      status: 1,
      healthLatency: 120,
      healthCheckedAt: new Date(now - 120000).toISOString(),
      remark: '',
      created: '2025-06-02T09:00:00Z',
      updated: '2025-06-09T11:00:00Z',
    },
    {
      mcpId: 3,
      name: 'postgres',
      displayName: 'PostgreSQL',
      icon: 'database',
      description: 'PostgreSQL 数据库查询和分析',
      transport: 'Stdio',
      httpUrl: '',
      httpHeader: '',
      httpProxyUrl: '',
      stdioType: 'npx',
      stdioEnv: '',
      stdioArgs: '["@modelcontextprotocol/server-postgres","postgresql://localhost:5432/mydb"]',
      status: 1,
      healthLatency: 80,
      healthCheckedAt: new Date(now - 300000).toISOString(),
      remark: '',
      created: '2025-06-03T10:00:00Z',
      updated: '2025-06-08T16:00:00Z',
    },
    {
      mcpId: 4,
      name: 'slack',
      displayName: 'Slack',
      icon: 'message-square',
      description: '发送和读取 Slack 消息',
      transport: 'Stdio',
      httpUrl: '',
      httpHeader: '',
      httpProxyUrl: '',
      stdioType: 'npx',
      stdioEnv: '',
      stdioArgs: '["@modelcontextprotocol/server-slack"]',
      status: 0,
      healthLatency: 0,
      healthCheckedAt: '',
      remark: '',
      created: '2025-06-04T08:00:00Z',
      updated: '2025-06-04T08:00:00Z',
    },
    {
      mcpId: 5,
      name: 'brave-search',
      displayName: 'Brave 搜索',
      icon: 'search',
      description: '通过 Brave Search API 搜索网页',
      transport: 'SSE',
      httpUrl: 'https://brave-mcp.example.com/sse',
      httpHeader: '{"Authorization":"Bearer sk-brave-xxx"}',
      httpProxyUrl: '',
      stdioType: 'npx',
      stdioEnv: '',
      stdioArgs: '',
      status: 1,
      healthLatency: 210,
      healthCheckedAt: new Date(now - 90000).toISOString(),
      remark: '',
      created: '2025-06-05T10:00:00Z',
      updated: '2025-06-07T09:00:00Z',
    },
    {
      mcpId: 6,
      name: 'time',
      displayName: '时间服务',
      icon: 'clock',
      description: '获取当前时间和时区转换',
      transport: 'Stdio',
      httpUrl: '',
      httpHeader: '',
      httpProxyUrl: '',
      stdioType: 'npx',
      stdioEnv: '',
      stdioArgs: '["@modelcontextprotocol/server-time"]',
      status: 1,
      healthLatency: 35,
      healthCheckedAt: new Date(now - 50000).toISOString(),
      remark: '',
      created: '2025-06-06T08:00:00Z',
      updated: '2025-06-06T08:00:00Z',
    },
    {
      mcpId: 7,
      name: 'puppeteer',
      displayName: 'Puppeteer',
      icon: 'globe',
      description: '无头浏览器，截图和网页抓取',
      transport: 'Stdio',
      httpUrl: '',
      httpHeader: '',
      httpProxyUrl: '',
      stdioType: 'npx',
      stdioEnv: '',
      stdioArgs: '["@modelcontextprotocol/server-puppeteer"]',
      status: 1,
      healthLatency: 340,
      healthCheckedAt: new Date(now - 180000).toISOString(),
      remark: '',
      created: '2025-06-07T08:00:00Z',
      updated: '2025-06-07T08:00:00Z',
    },
    {
      mcpId: 8,
      name: 'memory',
      displayName: '记忆存储',
      icon: 'brain',
      description: '持久化记忆和图谱知识',
      transport: 'Stdio',
      httpUrl: '',
      httpHeader: '',
      httpProxyUrl: '',
      stdioType: 'npx',
      stdioEnv: '',
      stdioArgs: '["@modelcontextprotocol/server-memory"]',
      status: 1,
      healthLatency: 55,
      healthCheckedAt: new Date(now - 200000).toISOString(),
      remark: '',
      created: '2025-06-08T08:00:00Z',
      updated: '2025-06-08T08:00:00Z',
    },
    {
      mcpId: 9,
      name: 'fetch',
      displayName: 'Web Fetch',
      icon: 'download',
      description: '抓取网页内容并转换为 Markdown',
      transport: 'StreamableHttp',
      httpUrl: 'https://fetch-mcp.example.com/mcp',
      httpHeader: '{"Authorization":"Bearer sk-xxxxx"}',
      httpProxyUrl: '',
      stdioType: 'npx',
      stdioEnv: '',
      stdioArgs: '',
      status: 1,
      healthLatency: 150,
      healthCheckedAt: new Date(now - 70000).toISOString(),
      remark: '',
      created: '2025-06-09T08:00:00Z',
      updated: '2025-06-09T08:00:00Z',
    },
    {
      mcpId: 10,
      name: 'cloudflare',
      displayName: 'Cloudflare',
      icon: 'cloud',
      description: '管理 Cloudflare Workers 和 KV',
      transport: 'SSE',
      httpUrl: 'https://cf-mcp.example.com/sse',
      httpHeader: '{"CF-API-Token":"cf-xxx"}',
      httpProxyUrl: '',
      stdioType: 'npx',
      stdioEnv: '',
      stdioArgs: '',
      status: 0,
      healthLatency: 0,
      healthCheckedAt: new Date(now - 3600000).toISOString(),
      remark: '',
      created: '2025-06-10T08:00:00Z',
      updated: '2025-06-10T15:00:00Z',
    },
    {
      mcpId: 11,
      name: 'everart',
      displayName: 'EverArt 图片生成',
      icon: 'image',
      description: '通过 EverArt API 生成 AI 图片',
      transport: 'SSE',
      httpUrl: 'https://everart-mcp.example.com/sse',
      httpHeader: '',
      httpProxyUrl: 'http://proxy.internal:8080',
      stdioType: 'npx',
      stdioEnv: '',
      stdioArgs: '',
      status: 1,
      healthLatency: 280,
      healthCheckedAt: new Date(now - 40000).toISOString(),
      remark: '',
      created: '2025-06-11T08:00:00Z',
      updated: '2025-06-11T08:00:00Z',
    },
    {
      mcpId: 12,
      name: 'sequential-thinking',
      displayName: '顺序思考',
      icon: 'lightbulb',
      description: '通过顺序思考增强推理能力',
      transport: 'Stdio',
      httpUrl: '',
      httpHeader: '',
      httpProxyUrl: '',
      stdioType: 'npx',
      stdioEnv: '',
      stdioArgs: '["@modelcontextprotocol/server-sequential-thinking"]',
      status: 1,
      healthLatency: 60,
      healthCheckedAt: new Date(now - 150000).toISOString(),
      remark: '',
      created: '2025-06-12T08:00:00Z',
      updated: '2025-06-12T08:00:00Z',
    },
  ]
}

export function generateRecommendMcps(): McpRecommendItem[] {
  return [
    {
      name: 'filesystem',
      displayName: '文件系统',
      icon: 'folder',
      description: '访问本地文件系统',
      transport: 'Stdio',
      httpUrl: '',
      stdioType: 'npx',
      stdioArgs: '["@modelcontextprotocol/server-filesystem","/tmp"]',
      installed: true,
      mcpId: 1,
      mcpStatus: 1,
    },
    {
      name: 'github',
      displayName: 'GitHub',
      icon: 'git-branch',
      description: '管理 GitHub 仓库和 PR',
      transport: 'Stdio',
      httpUrl: '',
      stdioType: 'npx',
      stdioArgs: '["@modelcontextprotocol/server-github"]',
      installed: true,
      mcpId: 2,
      mcpStatus: 1,
    },
    {
      name: 'postgres',
      displayName: 'PostgreSQL',
      icon: 'database',
      description: '数据库查询和分析',
      transport: 'Stdio',
      httpUrl: '',
      stdioType: 'npx',
      stdioArgs: '["@modelcontextprotocol/server-postgres","postgresql://localhost:5432/mydb"]',
      installed: true,
      mcpId: 3,
      mcpStatus: 1,
    },
    {
      name: 'brave-search',
      displayName: 'Brave 搜索',
      icon: 'search',
      description: '网页搜索',
      transport: 'SSE',
      httpUrl: 'https://brave-mcp.example.com/sse',
      stdioType: 'npx',
      stdioArgs: '[]',
      installed: true,
      mcpId: 5,
      mcpStatus: 1,
    },
    {
      name: 'slack',
      displayName: 'Slack',
      icon: 'message-square',
      description: '发送和读取 Slack 消息',
      transport: 'Stdio',
      httpUrl: '',
      stdioType: 'npx',
      stdioArgs: '["@modelcontextprotocol/server-slack"]',
      installed: true,
      mcpId: 4,
      mcpStatus: 0,
    },
    {
      name: 'puppeteer',
      displayName: 'Puppeteer',
      icon: 'globe',
      description: '无头浏览器，网页截图和抓取',
      transport: 'Stdio',
      httpUrl: '',
      stdioType: 'npx',
      stdioArgs: '["@modelcontextprotocol/server-puppeteer"]',
      installed: true,
      mcpId: 7,
      mcpStatus: 1,
    },
    {
      name: 'memory',
      displayName: '记忆存储',
      icon: 'brain',
      description: '持久化记忆和图谱知识',
      transport: 'Stdio',
      httpUrl: '',
      stdioType: 'npx',
      stdioArgs: '["@modelcontextprotocol/server-memory"]',
      installed: true,
      mcpId: 8,
      mcpStatus: 1,
    },
    {
      name: 'fetch',
      displayName: 'Web Fetch',
      icon: 'download',
      description: '抓取网页内容',
      transport: 'StreamableHttp',
      httpUrl: 'https://fetch-mcp.example.com/mcp',
      stdioType: 'npx',
      stdioArgs: '[]',
      installed: true,
      mcpId: 9,
      mcpStatus: 1,
    },
    {
      name: 'everart',
      displayName: 'EverArt',
      icon: 'image',
      description: 'AI 图片生成',
      transport: 'SSE',
      httpUrl: 'https://everart-mcp.example.com/sse',
      stdioType: 'npx',
      stdioArgs: '[]',
      installed: true,
      mcpId: 11,
      mcpStatus: 1,
    },
    {
      name: 'exa',
      displayName: 'Exa 搜索',
      icon: 'search',
      description: '互联网搜索',
      transport: 'SSE',
      httpUrl: 'https://exa-mcp.example.com/sse',
      stdioType: 'npx',
      stdioArgs: '[]',
      installed: false,
      mcpId: 0,
      mcpStatus: 0,
    },
    {
      name: 'sqlite',
      displayName: 'SQLite',
      icon: 'database',
      description: 'SQLite 数据库操作',
      transport: 'Stdio',
      httpUrl: '',
      stdioType: 'uvx',
      stdioArgs: '["mcp-server-sqlite"]',
      installed: false,
      mcpId: 0,
      mcpStatus: 0,
    },
    {
      name: 'weather',
      displayName: '天气查询',
      icon: 'cloud',
      description: '获取天气信息',
      transport: 'Stdio',
      httpUrl: '',
      stdioType: 'npx',
      stdioArgs: '["@h1deya/mcp-server-weather"]',
      installed: false,
      mcpId: 0,
      mcpStatus: 0,
    },
  ]
}

/* ============================================
   Async mock functions
   ============================================ */

export async function getMcps(params?: {
  page?: number
  pageSize?: number
  search?: string
  transport?: string
}): Promise<PaginatedResponse<AdminMcpItem>> {
  await listDelay()

  const store = ensureStore()
  let filtered = [...store]

  if (params?.search) {
    const q = params.search.toLowerCase()
    filtered = filtered.filter(
      (m) =>
        m.name.toLowerCase().includes(q) ||
        m.displayName.toLowerCase().includes(q),
    )
  }

  if (params?.transport && params.transport !== 'all') {
    filtered = filtered.filter((m) => m.transport === params.transport)
  }

  const page = params?.page ?? 1
  const pageSize = params?.pageSize ?? PAGE_SIZE
  const totalCount = filtered.length
  const totalPage = Math.max(1, Math.ceil(totalCount / pageSize))
  const start = (page - 1) * pageSize
  const list = filtered.slice(start, start + pageSize)

  return { list, totalPage, totalCount, page, pageSize }
}

export async function getMcpDetail(id: number): Promise<AdminMcpItem> {
  await itemDelay()

  const store = ensureStore()
  const item = store.find((m) => m.mcpId === id)
  if (!item) {
    throw new Error('MCP 端点不存在')
  }

  return { ...item }
}

export async function createMcp(req: {
  name: string
  displayName: string
  icon?: string
  description?: string
  transport: string
  httpUrl?: string
  httpHeader?: string
  httpProxyUrl?: string
  stdioType?: string
  stdioEnv?: string
  stdioArgs?: string
  remark?: string
}): Promise<AdminMcpItem> {
  await mutationDelay()

  const store = ensureStore()
  const now = new Date().toISOString()
  const newItem: AdminMcpItem = {
    mcpId: nextId(),
    name: req.name,
    displayName: req.displayName || req.name,
    icon: req.icon ?? 'plug',
    description: req.description ?? '',
    transport: req.transport,
    httpUrl: req.httpUrl ?? '',
    httpHeader: req.httpHeader ?? '',
    httpProxyUrl: req.httpProxyUrl ?? '',
    stdioType: req.stdioType ?? 'npx',
    stdioEnv: req.stdioEnv ?? '',
    stdioArgs: req.stdioArgs ?? '',
    status: 1,
    healthLatency: Math.floor(Math.random() * 300) + 20,
    healthCheckedAt: now,
    remark: req.remark ?? '',
    created: now,
    updated: now,
  }
  store.unshift(newItem)
  return { ...newItem }
}

export async function updateMcp(
  id: number,
  req: {
    displayName?: string
    icon?: string
    description?: string
    httpUrl?: string
    httpHeader?: string
    httpProxyUrl?: string
    stdioType?: string
    stdioEnv?: string
    stdioArgs?: string
    remark?: string
    status?: number
  },
): Promise<AdminMcpItem> {
  await mutationDelay()

  const store = ensureStore()
  const index = store.findIndex((m) => m.mcpId === id)
  if (index === -1) {
    throw new Error('MCP 端点不存在')
  }

  const item = store[index]
  const updated: AdminMcpItem = {
    ...item,
    displayName: req.displayName ?? item.displayName,
    icon: req.icon ?? item.icon,
    description: req.description ?? item.description,
    httpUrl: req.httpUrl ?? item.httpUrl,
    httpHeader: req.httpHeader ?? item.httpHeader,
    httpProxyUrl: req.httpProxyUrl ?? item.httpProxyUrl,
    stdioType: req.stdioType ?? item.stdioType,
    stdioEnv: req.stdioEnv ?? item.stdioEnv,
    stdioArgs: req.stdioArgs ?? item.stdioArgs,
    remark: req.remark ?? item.remark,
    status: req.status ?? item.status,
    updated: new Date().toISOString(),
  }
  store[index] = updated
  return { ...updated }
}

export async function toggleMcpStatus(
  id: number,
  status: number,
): Promise<AdminMcpItem> {
  await mutationDelay()

  const store = ensureStore()
  const index = store.findIndex((m) => m.mcpId === id)
  if (index === -1) {
    throw new Error('MCP 端点不存在')
  }

  store[index] = {
    ...store[index],
    status,
    updated: new Date().toISOString(),
  }
  return { ...store[index] }
}

export async function deleteMcp(id: number): Promise<void> {
  await mutationDelay()

  const store = ensureStore()
  const index = store.findIndex((m) => m.mcpId === id)
  if (index === -1) {
    throw new Error('MCP 端点不存在')
  }
  store.splice(index, 1)
}

export async function getRecommendMcps(): Promise<McpRecommendItem[]> {
  await listDelay()
  return generateRecommendMcps()
}

export async function healthCheck(): Promise<{ status: string }> {
  await heavyDelay()
  return { status: 'checking' }
}

File: backend/vo/mcp.go
package vo

import "time"

// AdminMCPListReq 管理员 MCP 列表请求
type AdminMCPListReq struct {
	Search    string `url:"search"`    // 名称模糊搜索
	Transport string `url:"transport"` // 传输类型筛选: Stdio / SSE / StreamableHttp
	Status    *uint8 `url:"status"`    // 状态筛选: 0禁用 1启用，nil不筛选
	Page      int    `url:"page"`      // 页码
	PageSize  int    `url:"pageSize"`  // 每页条数
}

// AdminMCPItem 管理员 MCP 列表条目（敏感字段脱敏）
type AdminMCPItem struct {
	McpId           int        `json:"mcpId"`           // 主键ID
	Name            string     `json:"name"`            // 内部标识名，英文
	DisplayName     string     `json:"displayName"`     // 展示名称
	Icon            string     `json:"icon"`            // 图标
	Description     string     `json:"description"`     // 功能描述
	Transport       string     `json:"transport"`       // 传输类型: Stdio / SSE / StreamableHttp
	HttpUrl         string     `json:"httpUrl"`         // SSE/HTTP 服务地址
	HttpHeader      string     `json:"httpHeader"`      // 请求头（脱敏）
	HttpProxyUrl    string     `json:"httpProxyUrl"`    // 代理地址
	StdioType       string     `json:"stdioType"`       // Stdio 执行器类型: npx / uvx
	StdioEnv        string     `json:"stdioEnv"`        // 环境变量（脱敏）
	StdioArgs       string     `json:"stdioArgs"`       // 启动参数
	Status          uint8      `json:"status"`          // 状态: 0禁用 1启用
	AlwaysOn        int        `json:"alwaysOn"`        // 始终启用：0关闭 1开启
	HealthLatency   int        `json:"healthLatency"`   // 健康检测延迟（毫秒），离线为0
	HealthCheckedAt *time.Time `json:"healthCheckedAt"` // 最后健康检测时间
	Remark          string     `json:"remark"`          // 管理员备注
	Created         time.Time  `json:"created"`         // 创建时间
	Updated         time.Time  `json:"updated"`         // 更新时间
}

// AdminMCPDetailRsp 管理员 MCP 详情响应（包含原始敏感字段，用于编辑回填）
type AdminMCPDetailRsp struct {
	McpId           int        `json:"mcpId"`           // 主键ID
	Name            string     `json:"name"`            // 内部标识名，英文，不可修改
	DisplayName     string     `json:"displayName"`     // 展示名称
	Icon            string     `json:"icon"`            // 图标
	Description     string     `json:"description"`     // 功能描述
	Transport       string     `json:"transport"`       // 传输类型: Stdio / SSE / StreamableHttp，不可修改
	HttpUrl         string     `json:"httpUrl"`         // SSE/HTTP 服务地址
	HttpHeader      string     `json:"httpHeader"`      // 请求头（原始值）
	HttpProxyUrl    string     `json:"httpProxyUrl"`    // 代理地址
	StdioType       string     `json:"stdioType"`       // Stdio 执行器类型: npx / uvx
	StdioEnv        string     `json:"stdioEnv"`        // 环境变量（原始值）
	StdioArgs       string     `json:"stdioArgs"`       // 启动参数
	Status          uint8      `json:"status"`          // 状态: 0禁用 1启用
	AlwaysOn        int        `json:"alwaysOn"`        // 始终启用：0关闭 1开启
	HealthLatency   int        `json:"healthLatency"`   // 健康检测延迟（毫秒），离线为0
	HealthCheckedAt *time.Time `json:"healthCheckedAt"` // 最后健康检测时间
	Remark          string     `json:"remark"`          // 管理员备注
	Created         time.Time  `json:"created"`         // 创建时间
	Updated         time.Time  `json:"updated"`         // 更新时间
}

// AdminCreateMCPReq 管理员创建 MCP 请求
// 前端 httpHeader/stdioEnv 传 map[string]string，后端转为存储格式
type AdminCreateMCPReq struct {
	Name         string            `json:"name" validate:"required"`        // 内部标识名，英文，全局唯一
	DisplayName  string            `json:"displayName" validate:"required"` // 展示名称
	Icon         string            `json:"icon"`                            // 图标
	Description  string            `json:"description"`                     // 功能描述
	Transport    string            `json:"transport" validate:"required"`   // 传输类型: Stdio / SSE / StreamableHttp
	HttpUrl      string            `json:"httpUrl"`                         // SSE/HTTP 服务地址
	HttpHeader   map[string]string `json:"httpHeader"`                      // 请求头，前端传 map
	HttpProxyUrl string            `json:"httpProxyUrl"`                    // 代理地址，非必填
	StdioType    string            `json:"stdioType"`                       // Stdio 执行器类型: npx / uvx
	StdioEnv     map[string]string `json:"stdioEnv"`                        // 环境变量，前端传 map
	StdioArgs    []string          `json:"stdioArgs"`                       // 启动参数，如 ["@modelcontextprotocol/server-filesystem", "/tmp"]
	Remark       string            `json:"remark"`                          // 管理员备注
}

// AdminUpdateMCPReq 管理员编辑 MCP 请求（仅传需要修改的字段，name/transport 不可改）
type AdminUpdateMCPReq struct {
	DisplayName  string            `json:"displayName"`  // 展示名称
	Icon         string            `json:"icon"`         // 图标
	Description  string            `json:"description"`  // 功能描述
	HttpUrl      string            `json:"httpUrl"`      // SSE/HTTP 服务地址
	HttpHeader   map[string]string `json:"httpHeader"`   // 请求头，前端传 map
	HttpProxyUrl string            `json:"httpProxyUrl"` // 代理地址
	StdioType    string            `json:"stdioType"`    // Stdio 执行器类型: npx / uvx
	StdioEnv     map[string]string `json:"stdioEnv"`     // 环境变量，前端传 map
	StdioArgs    []string          `json:"stdioArgs"`    // 启动参数
	Status       *uint8            `json:"status"`       // 状态: 0禁用 1启用，nil不修改
	Remark       string            `json:"remark"`       // 管理员备注
}

// AdminMCPToggleAlwaysOnReq 切换 MCP 始终启用请求
type AdminMCPToggleAlwaysOnReq struct {
	AlwaysOn int `json:"alwaysOn" validate:"oneof=0 1"` // 始终启用：0关闭 1开启
}

// MCPRecommendItem 推荐 MCP 条目
type MCPRecommendItem struct {
	Name        string `json:"name"`        // 内部标识名，用于判断安装状态
	DisplayName string `json:"displayName"` // 展示名称
	Icon        string `json:"icon"`        // 图标
	Description string `json:"description"` // 功能描述
	Transport   string `json:"transport"`   // 传输类型
	HttpUrl     string `json:"httpUrl"`     // SSE/HTTP 服务地址（推荐模板）
	HttpHeader  string `json:"httpHeader"`  // SSE/HTTP 请求头（JSON 字符串）
	StdioType   string `json:"stdioType"`   // Stdio 执行器类型
	StdioArgs   string `json:"stdioArgs"`   // 启动参数
	StdioEnv    string `json:"stdioEnv"`    // 环境变量（KEY=VALUE 数组字符串）
	Installed   bool   `json:"installed"`   // 是否已安装（name 匹配 deleted=0 的记录）
	McpId       int    `json:"mcpId"`       // 已安装时的 mcpId，0表示未安装
	McpStatus   uint8  `json:"mcpStatus"`   // 已安装时的状态: 0禁用 1启用
}

// MCPCodePreview MCP 代码配置预览，前端展示用
type MCPCodePreview struct {
	Type    string            `json:"type"`    // 传输类型小写: stdio / sse / http
	Command string            `json:"command"` // Stdio 命令: npx 或 uvx
	Args    []string          `json:"args"`    // Stdio 参数
	Url     string            `json:"url"`     // SSE/HTTP 服务地址
	Headers map[string]string `json:"headers"` // SSE/HTTP 请求头
	Env     map[string]string `json:"env"`     // 环境变量
}

// MCPRecommendDef 推荐 MCP 定义
type MCPRecommendDef struct {
	Name        string   // 内部标识名
	DisplayName string   // 展示名称
	Icon        string   // 图标
	Description string   // 功能描述
	Transport   string   // 传输类型: Stdio / SSE / StreamableHttp
	HttpUrl     string   // SSE/HTTP 服务地址
	StdioType   string   // Stdio 执行器类型: npx / uvx
	StdioArgs   []string // 启动参数
}

// UserMCPItem 用户可选 MCP 列表条目
type UserMCPItem struct {
	McpId       int    `json:"mcpId"`       // 主键ID
	Name        string `json:"name"`        // 内部标识名，英文
	DisplayName string `json:"displayName"` // 展示名称
	Icon        string `json:"icon"`        // 图标
	Description string `json:"description"` // 功能描述
}

// MCPHealthItem MCP 服务健康状态条目，用于系统信息页展示
type MCPHealthItem struct {
	McpId           int        `json:"mcpId"`           // 主键ID
	Name            string     `json:"name"`            // 内部标识名
	DisplayName     string     `json:"displayName"`     // 展示名称
	Icon            string     `json:"icon"`            // 图标
	HealthLatency   int        `json:"healthLatency"`   // 健康检测延迟（毫秒），离线为0
	HealthCheckedAt *time.Time `json:"healthCheckedAt"` // 最后健康检测时间
}
